src/rabbit_amqqueue.erl

```diff
@@ -274,9 +274,15 @@ declare(QueueName, Durable, AutoDelete, Args, Owner, Node) ->
                                       recoverable_slaves = [],
                                       gm_pids            = [],
                                       state              = live})),
-    Node = rabbit_mirror_queue_misc:initial_queue_node(Q, Node),
+
+    Node1 = case rabbit_queue_master_location_misc:get_location(Q)  of
+              {ok, Node0}  -> Node0;
+              {error, _}   -> Node
+            end,
+
+    Node1 = rabbit_mirror_queue_misc:initial_queue_node(Q, Node1),
```

Comment:
I think the key to understand that is the comment here: https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_amqqueue.erl#L262 

There are cases where the Master node might change. I suggest to see what's going on on the current version of rabbit vs. this branch when mirroring is in place and the `ha-params` policy is in place